presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -77,14 +78,20 @@ private PageBuilder(int initialExpectedEntries, int maxPageBytes, List<? extends
             }
         }
 
-        blockBuilders = new BlockBuilder[types.size()];
-        for (int i = 0; i < blockBuilders.length; i++) {
-            blockBuilders[i] = types.get(i).createBlockBuilder(
-                    pageBuilderStatus.createBlockBuilderStatus(),
-                    expectedEntries,
-                    pageBuilderStatus.getMaxBlockSizeInBytes() / expectedEntries);
-        }
         this.initialExpectedEntries = expectedEntries;
+
+        if (builders.isPresent()) {
+            blockBuilders = builders.get();
```

Comment:
@dain can you take another look at the PageBuilder constructor? I want to make sure I am passing the correct block builder status instances.